test/erlcloud_config_tests.erl

```diff
@@ -0,0 +1,513 @@
+    #{<<"nextToken">> => <<"string">>,
+        <<"resourceCounts">> => [
+            #{
+                <<"count">> => 10,
+                <<"resourceType">> => <<"string">>
+            }
+        ],
+        <<"totalDiscoveredResources">> => 1234567890
+    }
+).
+
+-define(GET_RESOURCE_CONFIG_HISTORY_RESP,
+    #{<<"chronologicalOrder">> => <<"string">>,
+        <<"earlierTime">> => 01234567890,
+        <<"laterTime">> => 01234567890,
+        <<"limit">> => 01234567890,
+        <<"nextToken">> => <<"string">>,
+        <<"resourceId">> => <<"string">>,
+        <<"resourceType">> => <<"string">>
+    }
+).
+
+-define(LIST_DISCOVERED_RESOURCES_RESP,
+    #{<<"nextToken">> => <<"string">>,
+        <<"resourceIdentifiers">> => [
+            #{<<"resourceDeletionTime">> => 01234567890,
+                <<"resourceId">> => <<"string">>,
+                <<"resourceName">> => <<"string">>,
+                <<"resourceType">> => <<"string">>
+            }
+        ]
+    }
+).
+
+setup() ->
+    erlcloud_config:configure("test-access-key", "test-secret-key"),
+    meck:new(?EHTTPC, [passthrough]),
+    meck:expect(?EHTTPC, request, 6, fun do_erlcloud_httpc_request/6),
+    [?EHTTPC].
+
+erlcloud_config_test_() ->
+    {
+        foreach,
+        fun setup/0,
+        fun meck:unload/1,
+        [
+            fun test_describe_compliance_by_config_rule/0,
+            fun test_describe_compliance_by_resource/0,
+            fun test_describe_config_rule_evaluation_status/0,
+            fun test_describe_config_rules/0,
+            fun test_describe_configuration_recorders/0,
+            fun test_describe_configuration_recorder_status/0,
+            fun test_describe_delivery_channels/0,
+            fun test_describe_delivery_channel_status/0,
+            fun test_get_compliance_details_by_config_rule/0,
+            fun test_get_compliance_details_by_resource/0,
+            fun test_get_compliance_summary_by_config_rule/0,
+            fun test_get_compliance_summary_by_resource_type/0,
+            fun test_get_discovered_resource_counts/0,
+            fun test_get_resource_config_history/0,
+            fun test_list_discovered_resources/0
+        ]
+    }.
+
+test_describe_compliance_by_config_rule() ->
+    Request = #{?COMPLIANCE_TYPES => ?TEST_COMPLIANCE_TYPES,
+        ?CONFIG_RULE_NAMES        => ?TEST_CONFIG_RULE_NAMES,
+        ?NEXT_TOKEN_LABEL         => <<"">>},
+    Expected = {ok, ?DESCRIBE_COMPLIANCE_BY_CONFIG_RULE_RESP, <<"">>},
+    TestFun  = fun() ->
+        erlcloud_config:describe_compliance_by_config_rule(Request)
+               end,
+    do_test(Request, Expected, TestFun).
+
+test_describe_compliance_by_resource() ->
+    Request = #{?COMPLIANCE_TYPES => ?TEST_COMPLIANCE_TYPES,
+        ?LIMIT                    => ?TEST_LIMIT,
+        ?RESOURCE_ID              => ?TEST_RESOURCE_ID,
+        ?RESOURCE_TYPE            => ?TEST_RESOURCE_TYPE,
+        ?NEXT_TOKEN_LABEL         => <<"">>},
```

Comment:
Interface concern:
It does not seen obvious to me to add empty token for the first query. Will the call work if I omit the next token?